model/artifact/artifact_file.go

```diff
@@ -84,7 +97,39 @@ func GetAllArtifacts(tasks []TaskIDAndExecution) ([]File, error) {
 		}
 	}
 	files := []File{}
+	catcher := grip.NewBasicCatcher()
 	for _, artifact := range artifacts {
+		for i := range artifact.Files {
+			if artifact.Files[i].Visibility == "signed" {
+				if artifact.Files[i].AwsSecret == "" || artifact.Files[i].AwsKey == "" || artifact.Files[i].Bucket == "" || artifact.Files[i].FileKey == "" {
+					err = errors.New("error presigning the url for artifact")
+					catcher.Add(err)
+				}
+
+				sess, err := session.NewSession(&aws.Config{
+					Region: aws.String("us-east-1"),
+					Credentials: credentials.NewStaticCredentialsFromCreds(credentials.Value{
+						AccessKeyID:     artifact.Files[i].AwsKey,
+						SecretAccessKey: artifact.Files[i].AwsSecret,
+					}),
+				})
+				if err != nil {
+					log.Fatalf("problem creating session: %s", err.Error())
```

Comment:
use grip for logging, and the fatal loggers are not safe to use in the application code.